effects/gh-pages/default.nix

```diff
@@ -0,0 +1,64 @@
+{ pkgs
+, runIf
+, mkEffect
+}:
+
+let
+  inherit (pkgs) lib;
+  inherit (pkgs.lib) optionalString;
+in
+
+{ gh-pages
+, branchName ? "gh-pages"
+, condition ? { ref, ... }: lib.elem ref [ "refs/heads/main" "refs/heads/master" ]
+, committer ? {
+    name = "Andrey Vlasov";
+    email = "andreyvlasov+gh-pages-builder@mlabs.city";
```

Comment:
This probably shouldn't be the default.
With the modular approach this probably wouldn't even have to be part of the gh pages interface.

Reply:
It'll be enough for internal usage in MLabs :) I would work on this if we were intending to merge this PR